apps/dead_letter/lib/dead_letter/server.ex

```diff
@@ -108,4 +110,12 @@ defmodule DeadLetter.Server do
   defp get_stacktrace({_, stacktrace}) when is_list(stacktrace) do
     stacktrace
   end
+
+  defp add_dead_letter_count(dataset_id, reason) do
```

Comment:
Can this raise an exception? If so, will it kill the whole dead_letter process? If possible, I feel that we should tolerate an exception here and log it. Updating the metrics is not a critical function.